File: saferide-backend-python/models/entities/parent_child_relationship.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from enum import Enum

class RelationshipType(str, Enum):
    PARENT = "parent"
    CHILD = "child"
    ESCORT = "escort"

class ParentChildRelationship(BaseModel):
    id: str = Field(..., description="Relationship ID")
    parent_id: str = Field(..., description="Parent user ID")
    child_id: str = Field(..., description="Child user ID")
    escort_id: Optional[str] = Field(None, description="Escort user ID (if assigned)")
    relationship_type: RelationshipType = Field(..., description="Type of relationship")
    is_active: bool = Field(True, description="Whether relationship is active")
    created_at: datetime = Field(default_factory=datetime.now)
    updated_at: datetime = Field(default_factory=datetime.now)
    notes: Optional[str] = Field(None, description="Additional notes about the relationship")

class ParentChildRelationshipCreate(BaseModel):
    parent_id: str = Field(..., description="Parent user ID")
    child_id: str = Field(..., description="Child user ID")
    escort_id: Optional[str] = Field(None, description="Escort user ID")
    relationship_type: RelationshipType = Field(..., description="Type of relationship")
    notes: Optional[str] = Field(None, description="Additional notes")

class ParentChildRelationshipUpdate(BaseModel):
    escort_id: Optional[str] = Field(None, description="Escort user ID")
    is_active: Optional[bool] = Field(None, description="Whether relationship is active")
    notes: Optional[str] = Field(None, description="Additional notes")

class UserRelationships(BaseModel):
    user_id: str = Field(..., description="User ID")
    as_parent: List[ParentChildRelationship] = Field(default_factory=list, description="Relationships where user is parent")
    as_child: List[ParentChildRelationship] = Field(default_factory=list, description="Relationships where user is child")
    as_escort: List[ParentChildRelationship] = Field(default_factory=list, description="Relationships where user is escort") 

File: saferide-frontend/src/models/interfaces/parent_child_relationship.ts
/**
 * Relationship type enum
 */
export enum RelationshipType {
  PARENT = 'parent',
  CHILD = 'child',
  ESCORT = 'escort'
}

/**
 * Parent-child relationship interface
 */
export interface ParentChildRelationship {
  id: string
  parent_id: string
  child_id: string
  escort_id?: string
  relationship_type: RelationshipType
  is_active: boolean
  created_at: string
  updated_at: string
  notes?: string
}

/**
 * Create relationship request interface
 */
export interface ParentChildRelationshipCreate {
  parent_id: string
  child_id: string
  escort_id?: string
  relationship_type: RelationshipType
  notes?: string
}

/**
 * Update relationship request interface
 */
export interface ParentChildRelationshipUpdate {
  escort_id?: string
  is_active?: boolean
  notes?: string
}

/**
 * User relationships interface showing all relationships for a user
 */
export interface UserRelationships {
  user_id: string
  as_parent: ParentChildRelationship[]
  as_child: ParentChildRelationship[]
  as_escort: ParentChildRelationship[]
} 